lighty-core/lighty-codecs-util/src/main/java/io/lighty/codecs/util/NodeConverter.java

```diff
@@ -7,61 +7,124 @@
  */
 package io.lighty.codecs.util;
 
+import io.lighty.codecs.util.exception.DeserializationException;
+import io.lighty.codecs.util.exception.SerializationException;
 import java.io.Reader;
 import java.io.Writer;
-import org.opendaylight.netconf.api.xml.XmlElement;
+import java.util.Optional;
 import org.opendaylight.yangtools.yang.common.QName;
+import org.opendaylight.yangtools.yang.data.api.YangInstanceIdentifier;
 import org.opendaylight.yangtools.yang.data.api.schema.NormalizedNode;
 import org.opendaylight.yangtools.yang.model.api.EffectiveModelContext;
 import org.opendaylight.yangtools.yang.model.api.SchemaNode;
+import org.opendaylight.yangtools.yang.model.api.stmt.SchemaNodeIdentifier;
+import org.opendaylight.yangtools.yang.model.util.SchemaInferenceStack;
 
 /**
  * This interface may be useful when (de)serializing {@link NormalizedNode}s (from)into its XML or
  * JSON representation. Currently there are two implementations {@link XmlNodeConverter} and
  * {@link JsonNodeConverter}.
  */
 public interface NodeConverter {
-    /**
-     * This method will serialize the given {@link NormalizedNode} into its string representation.
-     *
-     * @see ConverterUtils#getSchemaNode(EffectiveModelContext, QName)
-     * @param schemaNode parent schema node used during serialization
-     * @param normalizedNode normalized nodes to be serialized
-     * @return {@link Writer}
-     * @throws SerializationException may be throws while serializing data
-     */
-    Writer serializeData(SchemaNode schemaNode, NormalizedNode normalizedNode) throws SerializationException;
+
+    @Deprecated
+    default Writer serializeData(SchemaNode schemaNode, NormalizedNode normalizedNode) throws SerializationException {
+        return serializeData(SchemaInferenceStack.ofSchemaPath(getModelContext(), schemaNode.getPath()),
+                normalizedNode);
+    }
+
+    default Writer serializeData(SchemaNodeIdentifier.Absolute schemaNodeIdentifier, NormalizedNode normalizedNode)
+            throws SerializationException {
+        return serializeData(SchemaInferenceStack.of(getModelContext(), schemaNodeIdentifier), normalizedNode);
+    }
+
+    default Writer serializeData(NormalizedNode normalizedNode, QName... schemaIdentifierPath)
+            throws SerializationException {
+        return serializeData(SchemaNodeIdentifier.Absolute.of(schemaIdentifierPath), normalizedNode);
+    }
+
+    default Writer serializeData(NormalizedNode normalizedNode) throws SerializationException {
+        return serializeData(SchemaInferenceStack.of(getModelContext()), normalizedNode);
+    }
+
+    default Writer serializeData(YangInstanceIdentifier yangInstanceIdentifier, NormalizedNode normalizedNode)
```

Comment:
I mean.., why not cover more input types, if we can do that this easily. Tests are not necessary since all the defaults do is call appropriate ODL's construction of SchemaInferenceStack from that input parameter, no reason to test that.